### ISSUE TITLE
Document "outer click" alternatives in Accessibility section

### ISSUE THREAD

**Issue by gaearon** [reactjs/reactjs.org#849]
This comes up pretty often: https://mobile.twitter.com/ryanflorence/status/989940719451127808

Not to steal @ryanflorence’s thunder but I think we should have something on the Accessibility page that helps explain how to deal with this scenario.

cc @AlmeroSteyn

**Comment by AlmeroSteyn** [comment]
Really cool idea!

Would like to brainstorm a bit first about where best to put this and if other common gotchas should eventually end up in the docs too and in what form. Off the top of my head thinking of things like:

- Clickable `div`'s with no `tabindex`.
- `select`'s with onChange handlers.
- Infinite scroll.

These will all cause similar keyboard disturbances.

Inititially the take on the doc was to keep it lean and point to resources where devs can find this kind of info. But in these cases I think that explicitly stating the React specific solutions could help.

What is the feeling out there. Maybe a section right at the end of the doc with some common things to avoid?

cc @jessebeach

**Comment by AlmeroSteyn** [comment]
Got some greate inspiration this morning reading the release notes for 16.4 about where to put this in the a11y doc. Will be putting together a PR soon!

**Comment by AlmeroSteyn** [comment]
Pull request opened at #923 .
